From 7e5d7b62ef615ab9c24a6f519a3858623ad93c47 Mon Sep 17 00:00:00 2001
From: Ram Subramanian <ram@meraki.com>
Date: Mon, 15 Apr 2019 23:15:53 -0700
Subject: [PATCH] Check cache for squid_filedescriptors_num before setting it

If it already exists (say passed in as an argument to configure), no
need to re-calculate it.
---
 acinclude/os-deps.m4 | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/acinclude/os-deps.m4 b/acinclude/os-deps.m4
index c1cad52..1af766b 100644
--- a/acinclude/os-deps.m4
+++ b/acinclude/os-deps.m4
@@ -221,6 +221,7 @@ dnl sets shell var squid_filedescriptors_num
 
 AC_DEFUN([SQUID_CHECK_MAXFD],[
 AC_CHECK_FUNCS(setrlimit)
+AC_CACHE_CHECK([number of max file descriptors],[squid_filedescriptors_num],[
 AC_MSG_CHECKING(Maximum number of filedescriptors we can open)
 dnl damn! FreeBSD pthreads break dup2().
 SQUID_STATE_SAVE(maxfd)
@@ -301,6 +302,7 @@ int main(int argc, char **argv) {
   esac
   AC_MSG_RESULT($squid_filedescriptors_num)
 SQUID_STATE_ROLLBACK(maxfd)
+])
 
 if test `expr $squid_filedescriptors_num % 64` != 0; then
     AC_MSG_WARN([$squid_filedescriptors_num is not an multiple of 64. This may cause issues on certain platforms.])
-- 
2.21.0

